### PR TITLE
refactor: suppress unused variable warnings in release builds

### PR DESCRIPTION
**Pull request by gpanders**

### CODE REVIEW

**Comment by gpanders** [comment]
> Should we suppress this on cmake level instead?

If it's possible to suppress only variables that are unused because of an `assert` that was optimized out, then that would be great. But otherwise I think the unused variable warning is good to have.

**Review by dundargoc — APPROVED** [review]

**Comment by gpanders** [comment]
This change was also included in https://github.com/neovim/neovim/pull/26324, so https://github.com/neovim/neovim/commit/758fdc287d7a3382c22f6e1270a87e7f7fd3fadf is empty.

File: src/nvim/tui/input.c [hunk 1]
```diff
@@ -705,6 +705,7 @@ static void handle_raw_buffer(TermInput *input, bool force)
         const size_t bufsize = termkey_get_buffer_size(input->tk);
         const bool success = termkey_set_buffer_size(input->tk, MAX(bufsize + delta, bufsize * 2));
         assert(success);
+        (void)success;
```
Comment (justinmk) [inline src/nvim/tui/input.c:708]: Better to use abort https://github.com/neovim/neovim/pull/26309#discussion_r1410902728

Comment (zeertzjq) [reply src/nvim/tui/input.c:708]: I don't think it should abort in Release builds.

Comment (justinmk) [reply src/nvim/tui/input.c:708]: Why? If we fail to allocate, we always abort.

**Pull request merged**